Assets/_Functionalities/Wms/Scripts/WMSLayerGameObject.cs

```diff
@@ -21,18 +27,119 @@ public class WMSLayerGameObject : CartesianTileLayerGameObject, ILayerWithProper
         private WMSTileDataLayer wmsProjectionLayer;
         protected LayerURLPropertyData urlPropertyData = new();
 
+        [SerializeField] private GameObject legendPanelPrefab;
+        private static Legend legend;
+        [SerializeField] private Vector2Int legendOffsetFromParent;
+
         protected override void Awake()
         {
             base.Awake();
             wmsProjectionLayer = GetComponent<WMSTileDataLayer>();
+            LayerData.LayerSelected.AddListener(OnSelectLayer);
+            LayerData.LayerDeselected.AddListener(OnDeselectLayer);
         }
 
         protected override void Start()
         {
             base.Start();
             WMSProjectionLayer.WmsUrl = urlPropertyData.Data.ToString();
+            
             LayerData.LayerOrderChanged.AddListener(SetRenderOrder);
+
             SetRenderOrder(LayerData.RootIndex);
+
+            if (legend == null)
+            {
+                GameObject legendObject = Instantiate(legendPanelPrefab);
+                legend = legendObject.GetComponent<Legend>();
```

Comment:
is het niet handiger om legend een singleton te maken, ipv hier een referentie bij te houden naar de static instance?